src/comic-issue/comic-issue.controller.ts

```diff
@@ -94,13 +99,28 @@ export class ComicIssueController {
     @Param('id') id: string,
     @WalletEntity() wallet: Wallet,
   ): Promise<ComicIssueDto> {
-    const comicIssue = await this.comicIssueService.findOneProtected(
+    const comicIssue = await this.comicIssueService.findOne(
       +id,
       wallet.address,
     );
     return await toComicIssueDto(comicIssue);
   }
 
+  @Get('get/:id/pages')
+  async getPages(@Param('id') id: string): Promise<ComicPage[] | null> {
+    const pages = await this.comicPageService.getComicPagesForIssue(+id);
+    return sortBy(
+      await Promise.all(
+        pages.map(async (page) => ({
+          id: page.id,
+          pageNumber: page.pageNumber,
+          image: await getReadUrl(page.image),
+        })),
+      ),
+      'pageNumber',
+    ) as ComicPage[];
+  }
+
```

Comment:
This is mostly fine, as comic pages should be sorted. I would only move this logic to some DTO, helper function or some service as I don't want any calculations done in controllers.

I think you should work this in a `ComicPageDto`? Especially because you're casting here to `ComicPage` an object which should be an instance of `ComicPageDto

Reply:
Yes makes sense, totally agree with you.